src/scippneutron/file_loading/_transformations.py

```diff
@@ -85,30 +88,26 @@ def get_full_transformation_matrix(group: Group, nexus: LoadFromNexus) -> sc.Dat
     _get_transformations(depends_on, transformations, group, nexus.get_name(group),
                          nexus)
 
-    total_transform = sc.DataArray(
-        data=sc.spatial.affine_transforms(dims=["time"],
-                                          values=[np.identity(4)],
-                                          unit=sc.units.m),
-        coords={
-            "time":
-            sc.array(dims=["time"],
-                     values=[np.datetime64('1970-01-01T00:00:00', 'ns')],
-                     unit=sc.units.ns,
-                     dtype=sc.DType.datetime64)
-        })
+    total_transform = sc.spatial.affine_transform(value=np.identity(4), unit=sc.units.m)
 
     for transform in transformations:
-
-        xnew = sc.datetimes(values=np.unique(
-            sc.concat([
-                total_transform.coords["time"].to(unit=sc.units.ns, copy=True),
-                transform.coords["time"].to(unit=sc.units.ns, copy=True),
-            ],
-                      dim="time").values),
-                            dims=["time"])
-
-        total_transform = _interpolate_transform(
-            transform, xnew) * _interpolate_transform(total_transform, xnew)
+        if isinstance(total_transform, sc.DataArray):
+            xnew = sc.datetimes(values=np.unique(
+                sc.concat([
+                    total_transform.coords["time"].to(unit=sc.units.ns, copy=True),
+                    transform.coords["time"].to(unit=sc.units.ns, copy=True),
+                ],
+                          dim="time").values),
+                                dims=["time"])
+            total_transform = _interpolate_transform(
+                transform, xnew) * _interpolate_transform(total_transform, xnew)
+        else:
+            if isinstance(transform, sc.DataArray):
+                xnew = transform.coords["time"]
+                total_transform = _interpolate_transform(
+                    transform, xnew) * _interpolate_transform(total_transform, xnew)
```

Comment:
Why do we need to interpolate `total_transform`? If I understand correctly it is a variable (without time dim), so shouldn't this just be something like:
```suggestion
            if isinstance(transform, sc.DataArray):
                xnew = transform.coords["time"]
                total_transform = _interpolate_transform(transform, xnew) * total_transform
```
would that also obviate the branch in `_interpolate_transform` above?